lib/datasets/estatjp.rb

```diff
@@ -0,0 +1,302 @@
+                            app_id,
+                            stats_data_id,
+                            area: nil, cat: nil, time: nil)
+        # generates url for query
+        params = {
+          appId: app_id, lang: 'J',
+          statsDataId: stats_data_id, # 統計表ID
+          metaGetFlg: 'Y', cntGetFlg: 'N',
+          sectionHeaderFlg: '1'
+        }
+        # cdArea: ["01105", "01106"].join(","), # 地域事項
+        params['cdArea'] = area.join(',') if area.instance_of?(Array)
+        # cdCat01: ["A2101", "A210101", "A210102", "A2201", "A2301", "A4101", "A4200", "A5101", "A5102"].join(","), # 分類事項
+        params['cdCat01'] = cat.join(',') if cat.instance_of?(Array)
+        # cdTime: ["1981100000", "1982100000" ,"1984100000"].join(","), # 時間軸事項
+        params['cdTime'] = time.join(',') if time.instance_of?(Array)
+
+        URI.parse("#{base_url}?#{URI.encode_www_form(params)}")
+      end
+
+      def self.extract_def(data, id)
+        rec = data['GET_STATS_DATA']['STATISTICAL_DATA']\
+        ['CLASS_INF']['CLASS_OBJ']
+        rec.select { |x| x['@id'] == id }
+      end
+
+      def self.index_def(data_def)
+        unless data_def.first['CLASS'].instance_of?(Array)
+          # convert to array when number of element is 1
+          data_def.first['CLASS'] = [data_def.first['CLASS']]
+        end
+        Hash[*data_def.first['CLASS'].map { |x| [x['@code'], x] }.flatten]
+      end
+
+      def self.get_values(data)
+        data['GET_STATS_DATA']['STATISTICAL_DATA']['DATA_INF']['VALUE']
+      end
+
+      #
+      # generate accessor instance for e-Stat API.
+      # for detail spec : https://www.e-stat.go.jp/api/api-info/e-stat-manual
+      # @param [String] api_version API Version (defaults to `'2.1'`)
+      # @param [String] stats_data_id 統計表ID
+      # @param [Array<String>] cat 分類事項 (省略時はすべて取得)
+      # @param [Array<String>] area 地域事項 (省略時はすべて取得)
+      # @param [Array<String>] time 時間軸事項 (省略時はすべて取得)
+      # @param [Array<Number>] skip_level 省略する階層レベル (defaults to `[1]`)
+      # @param [String] hierarchy_selection 階層関係の上位・末端のフィルタリングをする ('child', 'parent', 'both') (例: 札幌市○○区 -> 'child':札幌市○○区のみ; 'parent':札幌市のみ; 'both':フィルタリングしない)
+      # @param [Boolean] skip_nil_column 1行でも欠損がある列をスキップする
+      # @param [Boolean] skip_nil_row 1列でも欠損がある行をスキップする
+      # @example
+      #   estat = Datasets::Estatjp::JSONAPI.new(
+      #     "0000020201", # Ａ　人口・世帯
+      #     cat: ["A1101"], # A1101_人口総数
+      #     area: ["01105", "01106"], # "北海道 札幌市 豊平区", "北海道 札幌市 南区"
+      #     time: ["1981100000", "1982100000"],
+      #     hierarchy_selection: 'child' , # 例: 札幌市○○区があるときは札幌市をスキップ
+      #     skip_child_area: true, # 例: 札幌市○○区をスキップして札幌市を残す TODO skip_(parent|child) を統合する
+      #     skip_nil_column: true, #  1行でも欠損があったら列をスキップする
+      #     skip_nil_row: false, # 1列でも欠損があったら行をスキップする
+      #   )
+      #
+      def initialize(stats_data_id,
+                     api_version: '2.1',
```

Comment:
I think that we should not parameterize `api_version` here because response format may be changed by different `api_version`.
If response format is changed, our code will be broken.

Reply:
I've been wondering about future compatibility for e-Stat API. This time, I'll make this value fixed as your suggestion since we don't know their update schedule of API.